x/ecocredit/basket/msg_create.go

```diff
@@ -57,8 +57,17 @@ func (m MsgCreate) ValidateBasic() error {
 }
 
 // Validate additional validation with access to the state data.
-func (m MsgCreate) Validate(minFee sdk.Coin) error {
-	// TODO: update function to do stateful validation
+// minFee must be sorted.
+func (m MsgCreate) Validate(minFee sdk.Coins) error {
+	// Note if user is adding too much denoms then we fails as well.
+	if len(minFee) != len(m.Fee) {
+		return errBadReq.Wrapf("Wrong denom set, expected %v", m.Fee)
```

Comment:
aliasing `sdkerrors.ErrInvalidateRequest` with `errBadReq` does not enhance readability IMHO